docs/report_stage_iv.md

```diff
@@ -0,0 +1,137 @@
+
+## Introduction and project scope
+This project continues the previous stages of the Gas Cost Estimator. Please visit https://github.com/imapp-pl/gas-cost-estimator to find more information. After publishing our report from the second and third stages of the Gas Cost Estimator project we received feedback from the community. The community expressed the need to see other implementations being included in the research as well as to have the tooling automated and the benchmarks standardized.
+
+### EVM Implementaions
+The following EVM implementations have been included in the research:
+- [EvmOne *](https://github.com/imapp-pl/evmone), version 0.12.0, commit [bf6c91d](https://github.com/ethereum/evmone/commit/bf6c91df5a0e0ad9de525eca8e9d8286c8ec5ab1)
+- [Go Ethereum](https://github.com/ethereum/go-ethereum), version  Rayingri (v1.14.8)
+, commit [a9523b6](https://github.com/ethereum/go-ethereum/commit/a9523b6428238a762e1a1e55e46ead47630c3a23)
+- [Erigon](https://github.com/erigontech/erigon), version 2.60.6, commit [d24e5d4](https://github.com/erigontech/erigon/commit/d24e5d45755d7b23075c507ad9216e1d60ad03de)
+- [EthereumJS](https://github.com/imapp-pl/ethereumjs-monorepo), version 8.1.0, commit [db8c0db](https://github.com/imapp-pl/ethereumjs-monorepo/commit/db8c0dbe76b366edae7b609960bd99ff00e10cf7)
+- [Nethermind](https://github.com/imapp-pl/nethermind), version 1.28.0, commit [e65c1cd](https://github.com/imapp-pl/nethermind/commit/e65c1cd59d858e3add8203d1bc21ec1c1f38de5b)
+- [Revm](https://github.com/imapp-pl/revm), version 13.0.0, commit [d82c762](https://github.com/imapp-pl/revm/commit/d82c7621d7256d8067e97845411c68a3e1d6ef57)
+- [Besu *](https://github.com/imapp-pl/besu)
+
+### Measured OPCODEs and precompiles
+In this stage, we finally measure all* OPCODEs together with the precompiles.
+
+### Tooling and automation *
+The reproducibility is the key to the research. We have provided a complete setup guide and tooling to make the execution of the benchmarks as easy as possible. There are two ways to perform the measurements in your environment. The first one is to use the provided scripts to build the EVM implementations and run the benchmarks. The second one is to use the provided release with precompiled binaries. The binaries are available for Linux x64, MacOS x64 and Windows.
+
+## Methodology
+
+### Measurement approach
+Our approach is to test each EVM implementation in isolation. That means that any host objects, storage access and other infrastructure elements are either mocked, or a minimal implementation is used. This allows us to measure the execution time of the OPCODEs directly, without any additional overhead.
```

Comment:
Something is not right . I have a feeling that each OPCODE is measured as one execution, independently. Maybe this additional sentence will clarify:
Measuring a single OPCODE execution time is inaccurate and susceptible to external factors. For that reason larger bytecode programs are executed and OPCODEs costs are estimated using statistic tools.

Reply:
Added explanation